# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/flyrank-bih/flyrank-ml-internship-starter/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [2]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

df = pd.read_csv("ai_student_impact_dataset.csv")
df["GPA_Change"] = df["Post_Semester_GPA"] - df["Pre_Semester_GPA"]

print("rows, cols:", df.shape)
print("columns:", list(df.columns))


rows, cols: (50000, 17)
columns: ['Student_ID', 'Major_Category', 'Year_of_Study', 'Pre_Semester_GPA', 'Weekly_GenAI_Hours', 'Primary_Use_Case', 'Prompt_Engineering_Skill', 'Tool_Diversity', 'Paid_Subscription', 'Traditional_Study_Hours', 'Perceived_AI_Dependency', 'Institutional_Policy', 'Anxiety_Level_During_Exams', 'Post_Semester_GPA', 'Skill_Retention_Score', 'Burnout_Risk_Level', 'GPA_Change']


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [3]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
burnout_counts = df["Burnout_Risk_Level"].value_counts()
print(burnout_counts)
print(f"\nShare labeled High burnout risk: {burnout_counts['High'] / len(df):.1%}")

Burnout_Risk_Level
Medium    21144
Low       16369
High      12487
Name: count, dtype: int64

Share labeled High burnout risk: 25.0%


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [3]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [4]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
overall_decline_rate = (df["GPA_Change"] < 0).mean()
decline_by_burnout = df.groupby("Burnout_Risk_Level")["GPA_Change"].apply(lambda g: (g < 0).mean())

print(f"Overall GPA-decline rate: {overall_decline_rate:.1%}")
print(decline_by_burnout.sort_values(ascending=False))

Overall GPA-decline rate: 12.4%
Burnout_Risk_Level
High      0.147914
Low       0.120105
Medium    0.112514
Name: GPA_Change, dtype: float64


In [5]:
# Number 2: does weekly GenAI usage actually differ across burnout risk groups?
genai_by_burnout = df.groupby("Burnout_Risk_Level")["Weekly_GenAI_Hours"].mean()
print(genai_by_burnout.sort_values(ascending=False))

Burnout_Risk_Level
High      15.214823
Medium     7.349047
Low        4.643648
Name: Weekly_GenAI_Hours, dtype: float64


In [6]:
# Number 3: what correlates with GPA change at all? (sanity check the signal exists)
corr = df[[
    "Weekly_GenAI_Hours", "Traditional_Study_Hours",
    "Perceived_AI_Dependency", "Anxiety_Level_During_Exams",
    "Skill_Retention_Score", "GPA_Change"
]].corr()["GPA_Change"].sort_values(ascending=False)
print(corr)

GPA_Change                    1.000000
Traditional_Study_Hours       0.376317
Skill_Retention_Score         0.195833
Perceived_AI_Dependency      -0.039342
Anxiety_Level_During_Exams   -0.040420
Weekly_GenAI_Hours           -0.046479
Name: GPA_Change, dtype: float64


### What these numbers say

- Overall, about **12.4%** of students finish the semester with a lower GPA than they started with. Among students already labeled **High Burnout Risk**, that rate increases to **14.8%**, compared with **11.3%** for the Medium group. This suggests there is a meaningful subgroup that may benefit from earlier support.

- Students in the **High Burnout Risk** group use GenAI for an average of **15.2 hours per week**, compared with **4.6 hours per week** for the Low Burnout Risk group—more than **three times higher**. This indicates that AI usage patterns may provide useful predictive information.

- Traditional study hours have the strongest positive relationship with GPA change (**correlation ≈ 0.38**), while GenAI usage hours and perceived AI dependency show only weak negative correlations (**≈ -0.04 to -0.05**). These are modest associations, so the project focuses on prediction rather than claiming that AI usage causes burnout.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, "predicting Google").*

- **I can say:** In this dataset, high GenAI usage and high burnout risk tend to occur together, and students with higher burnout risk show a somewhat higher rate of GPA decline. These patterns can serve as an early-warning signal for decision support, but they do not explain why the relationship exists.

- **I cannot say:** GenAI use *causes* burnout or GPA decline. This dataset is observational, based on self-reported responses collected over a single semester. Students who are already struggling may simply rely on GenAI more, which could produce the same correlation.

- **I cannot say:** These findings automatically generalize to students at other universities, academic programs, or semesters. External validation on independent datasets would be required before making broader claims.

- The observed correlations (approximately **-0.04 to -0.05**) are relatively weak. Any predictive model should therefore be compared against a simple baseline, and its value should be judged by whether it provides a meaningful improvement in decision support rather than by claiming perfect or highly accurate predictions.

In [10]:
# Quick sanity check I'll keep coming back to: the effect sizes are small, not dramatic.
print(corr[["Weekly_GenAI_Hours", "Perceived_AI_Dependency"]])

Weekly_GenAI_Hours        -0.046479
Perceived_AI_Dependency   -0.039342
Name: GPA_Change, dtype: float64


## Self-check

Before you submit, confirm each line honestly:

- [✅] Every section above is filled — markdown thinking AND the code that backs it
- [✅] The notebook runs top to bottom with no errors (Runtime → Run all)
- [✅] No client names, URLs, or private queries anywhere
- [✅] My claims use careful words: observed, measured, directional, decision-support
- [✅] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.

**Lane status:** provisional — freestyle, using `ai_student_impact_dataset.csv`. Open to revisiting by end of Week 4 if a predefined lane fits better or these effect sizes turn out too weak to build on.
